### DISCUSSION TITLE
Generator tracing: how to generate JavaScript with source maps from langium-lox?

### DISCUSSION THREAD

**Discussion by trusktr** [eclipse-langium/langium#1087] [Q&A]
For example, suppose we take langium-lox: and instead of making an interpreter we generate JavaScript code along with a source map. How would we do that with generator tracing?

**Comment by msujew** [comment]
Hey @trusktr,

there are a lot of tracing utilities available on our [generator interfaces](https://github.com/langium/langium/blob/0cd59e78a0b8e1dd681a379df2f1c06debee514a/packages/langium/src/generator/generator-node.ts#L76-L816). We don't offer direct integration with something like the [source-map](https://www.npmjs.com/package/@parcel/source-map) package, but you should be able to create a small glue layer between the package and our [`TraceRegion`](https://github.com/langium/langium/blob/0cd59e78a0b8e1dd681a379df2f1c06debee514a/packages/langium/src/generator/generator-tracing.ts#L39-L43) to get something working.

**Comment by sailingKieler** [comment]
Hey @trusktr,

yesterday I published the 2nd part of my blog series on generation, see https://www.typefox.io/blog/code-generation-for-langium-based-dsls-2. Generator tracing will be handled in the 3rd part, which I'm almost done with, and which is coming soon. I prepared a little demo project at https://github.com/langium/langium-in-browser-codegen-example/.
In my personal fork you already find the code demoing the creation of source maps based on tracing data, but sshhhhh! 🤫
